File: src/main/scala/br/com/fabioaraujo/calcula/produto/controller/ItemController.scala
package br.com.fabioaraujo.calcula.produto.controller

import br.com.fabioaraujo.calcula.produto.dto.Item
import br.com.fabioaraujo.calcula.produto.service.ItemService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._

@RestController
@RequestMapping(path = Array("item"))
class ItemController(@Autowired val itemService: ItemService) {

  @PostMapping(path = Array("criar"))
  @ResponseBody
  def criar(@RequestBody item: Item): Item = {
    val id = itemService.createItem(item)
    item.setId(id)
    item
  }

  @GetMapping(path = Array("getById"))
  @ResponseBody
  def getItem(@RequestParam id: Long): Item = {
    itemService.getItem(id)
  }

  @GetMapping(path = Array("listar"))
  @ResponseBody
  def listar(): java.lang.Iterable[Item] = {
    itemService.listItens()
  }
}

File: src/main/java/br/com/fabioaraujo/calcula/produto/controller/ItemController.java
package br.com.fabioaraujo.calcula.produto.controller;

import java.net.URI;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
import org.springframework.web.util.UriComponents;

import br.com.fabioaraujo.calcula.produto.dto.Item;
import br.com.fabioaraujo.calcula.produto.service.ItemService;

@RestController
@RequestMapping("itens")
class ItemController {

	@Autowired 
	private ItemService itemService;

	@PostMapping()
	@ResponseBody
	public ResponseEntity<Item> criar(@RequestBody Item item) {
		item.setId(null);
		Long id = itemService.save(item);
		item.setId(id);
		UriComponents uriComponents = MvcUriComponentsBuilder.fromMethodName(
				ItemController.class, "getItem", id).build();
		URI uri = uriComponents.toUri();
		return ResponseEntity.created(uri).build();
	}

	@PutMapping()
	@ResponseBody
	public ResponseEntity<Item> atualizar(@RequestBody Item item) {
		if(item.getId() == null) {
			return ResponseEntity.badRequest().build();
		}
		Long id = itemService.save(item);
		item.setId(id);
		return ResponseEntity.noContent().build();
	}

	@GetMapping("/{id}")
	@ResponseBody
	public ResponseEntity<Item> getItem(@PathVariable("id") Long id) {
		return ResponseEntity.ok(itemService.getItem(id));
	}

	@GetMapping()
	@ResponseBody
	public ResponseEntity<Iterable<Item>> listar(){
		return ResponseEntity.ok(itemService.listItens());
	}
}